include/deal.II/numerics/vector_tools.h

```diff
@@ -763,22 +762,31 @@ namespace VectorTools
    * values are disabled. The ordering of arguments to this function is such
    * that you need not give a second quadrature formula if you don't want to
    * project to the boundary first, but that you must if you want to do so.
-   *
-   * This function needs the mass matrix of the finite element space on the
-   * present grid. To this end, the mass matrix is assembled exactly using
-   * MatrixTools::create_mass_matrix. This function performs numerical
-   * quadrature using the given quadrature rule; you should therefore make
-   * sure that the given quadrature formula is also sufficient for the
-   * integration of the mass matrix.
+
+   * A MatrixFree implementation is used if the following consitions are met:
+   * - @p enforce_zero_boundary is false,
+   * - @p project_to_boundary_first is false,
+   * - the FiniteElement is supported by the MatrixFree class,
+   * - the FiniteElement has less than five components
+   * - the degree of the FiniteElement is less than nine.
+   * - dim==spacedim
+   *
+   * In this case, this function performs numerical quadrature for integration
+   * of the provided function while a QGauss(fe_degree+2) object is used for
```

Comment:
do you mean to say that "numerical quadrature USING THE GIVEN QUADRATURE FORMULA for integration..."?